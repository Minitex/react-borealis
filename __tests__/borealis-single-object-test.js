import React from 'react';
import renderer from 'react-test-renderer';
import Borealis from '../src/react-borealis';

jest.mock('react-openseadragon', () => 'OpenSeadragonViewer');

describe('Borealis', () => {
  it('should render correctly', () => {
    const config = {
      image: {
        thumbnail: 'https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg',
        type: 'image',
        label: 'Image',
        include_controls: true,
        sequenceMode: true,
        showReferenceStrip: true,
        defaultZoomLevel: 0,
        tileSources: [
          'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json',
          'https://ids.lib.harvard.edu/ids/iiif/25286610/info.json',
        ],
        transcript: {
          texts: ['First Image Item Transcript','Second Image Item  Transcript'],
          label: 'Image',
        },
        tocs: [
          'A Statue',
        ],
      },
    };

    const component = renderer.create(<Borealis config={config} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
