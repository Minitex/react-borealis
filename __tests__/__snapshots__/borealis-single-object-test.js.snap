exports[`Borealis should render correctly 1`] = `
<div>
  <div>
    <span />
    <div
      className="row image-nav">
      <ul
        className="nav nav-pills">
        <li>
          <a
            className="active transcript-nav-link"
            href="#/image/0"
            onClick={[Function]}
            style={Object {}}>
            Image
          </a>
        </li>
        <li>
          <a
            className="transcript-nav-link"
            href="#/image/0/transcript"
            onClick={[Function]}
            style={undefined}>
            Transcript
          </a>
        </li>
      </ul>
    </div>
    <OpenSeadragonViewer
      WrappedComponent="OpenSeadragonViewer"
      basename={undefined}
      config={
        Object {
          "defaultZoomLevel": 0,
          "include_controls": true,
          "label": "Image",
          "sequenceMode": true,
          "showReferenceStrip": true,
          "thumbnail": "https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg",
          "tileSources": Array [
            "https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json",
            "https://ids.lib.harvard.edu/ids/iiif/25286610/info.json",
          ],
          "tocs": Array [
            "A Statue",
          ],
          "transcript": Object {
            "label": "Image",
            "texts": Array [
              "First Image Item Transcript",
              "Second Image Item  Transcript",
            ],
          },
          "type": "image",
        }
      }
      defaultZoomLevel={0}
      getThumbnailPath={[Function]}
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/image/0",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      include_controls={true}
      label="Image"
      location={
        Object {
          "hash": "",
          "pathname": "/image/0",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": "0",
          },
          "path": "/image/:id",
          "url": "/image/0",
        }
      }
      sequenceMode={true}
      showReferenceStrip={true}
      staticContext={undefined}
      thumbnail="https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg"
      tileSources={
        Array [
          "https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json",
          "https://ids.lib.harvard.edu/ids/iiif/25286610/info.json",
        ]
      }
      tocs={
        Array [
          "A Statue",
        ]
      }
      transcript={
        Object {
          "label": "Image",
          "texts": Array [
            "First Image Item Transcript",
            "Second Image Item  Transcript",
          ],
        }
      }
      type="image" />
  </div>
</div>
`;
